lib/rules/no-restricted-amd.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * @file    Rule to ensure that required paths are ordered alphabetically
+ * @author  Ondřej Brejla <ondrej@brejla.cz>
+ * @author  Casey Visco <cvisco@gmail.com>
+ */
+
+"use strict";
+
+const ignore = require("ignore");
+const rjs = require("../utils/rjs");
+
+const isAmdDefine = rjs.isAmdDefine;
+const isAmdRequire = rjs.isAmdRequire;
+const getModuleFunction = rjs.getModuleFunction;
+const getDependencyStringNodes = rjs.getDependencyStringNodes;
+
+// -----------------------------------------------------------------------------
+// Configuration
+// -----------------------------------------------------------------------------
+
+const docs = {
+    description: "Ensure that required paths are ordered alphabetically",
+    category: "Stylistic Choices",
+    recommended: false,
+    url:
+        "https://github.com/cvisco/eslint-plugin-requirejs/blob/master/docs/rules/no-restricted-amd.md"
+};
+
+const arrayOfStrings = {
+    type: "array",
+    items: { type: "string" },
+    uniqueItems: true
+};
+
+const arrayOfStringsOrObjects = {
+    type: "array",
+    items: {
+        anyOf: [
+            { type: "string" },
+            {
+                type: "object",
+                properties: {
+                    name: { type: "string" },
+                    message: {
+                        type: "string",
+                        minLength: 1
+                    }
+                },
+                additionalProperties: false,
+                required: ["name"]
+            }
+        ]
+    },
+    uniqueItems: true
+};
+
+const schema = {
+    anyOf: [
+        arrayOfStringsOrObjects,
+        {
+            type: "array",
+            items: {
+                type: "object",
+                properties: {
+                    paths: arrayOfStringsOrObjects,
+                    patterns: arrayOfStrings
+                },
+                additionalProperties: false
+            },
+            additionalItems: false
+        }
+    ]
+};
+
+const DEFAULT_MESSAGE_TEMPLATE =
+    "'{{moduleName}}' module is restricted from being used.";
+const CUSTOM_MESSAGE_TEMPLATE =
+    "'{{moduleName}}' module is restricted from being used. {{customMessage}}";
+
+// -----------------------------------------------------------------------------
+// Rule Definition
+// -----------------------------------------------------------------------------
+
+function create(context) {
+    const options = Array.isArray(context.options) ? context.options : [];
+    const isPathAndPatternsObject =
+        typeof options[0] === "object" &&
+        (Object.prototype.hasOwnProperty.call(options[0], "paths") ||
+            Object.prototype.hasOwnProperty.call(options[0], "patterns"));
+
+    const restrictedPaths =
+        (isPathAndPatternsObject ? options[0].paths : context.options) || [];
```

Comment:
Actually it doesn't matter, because even `context.options` is `undefined`, `restrictedPaths` will be an empty array. However, for readability I'll change it. Good spot!